### DISCUSSION TITLE
play sbt run提示java.lang.NullPointerException

### DISCUSSION THREAD

**Discussion by xiaofeihzx** [playframework/playframework#12537] [Q&A]
![1713836924333(1)](https://github.com/playframework/playframework/assets/167833105/16b8d6a1-1402-4436-9ca6-3ec7f67d2017)
![1713836975664](https://github.com/playframework/playframework/assets/167833105/7dbec2be-16bf-40e6-8a22-8105a89871df)

**Comment by mkurz** [comment]
Please provide more information. Please post your `project/plugins.sbt` and `project/build.properties` file.
Also it looks like you use a very old Play version because you still use Scala 2.11.

**Reply by xiaofeihzx** [reply]
![Uploading 1713921324218.jpg…]()

**Reply by xiaofeihzx** [reply]
我的java 版本是java version "21.0.2". sbt版本1.9.9。  scala版本是2.11.7

**Reply by mkurz** [reply]
I doubt that Scala 2.11 will work correctly with Java 21, you should upgrade to Scala 2.13.
Also this sounds you use an old Play version, which one is it? You find that in your `project/plugins.sbt`.
Please copy/paste and do not post screenshots. Thanks!

**Reply by xiaofeihzx** [reply]
addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.5.10")  play是2.5.10版本

**Reply by mkurz** [reply]
Play 2.5 only supports Java 8 (not even 11) so this could be the reason you are getting those errors. Either use Java 8 or upgrade to latest Play 3.0, step by step, first upgrading to Play 2.6 then to 2.7 and so on, start here:
- https://www.playframework.com/documentation/latest/Highlights26
- https://www.playframework.com/documentation/latest/Migration26
